base/io.jl

```diff
@@ -888,11 +888,21 @@ function write(io::IO, s::Symbol)
 end
 
 function write(to::IO, from::IO)
-    n = 0
+    n::Int64 = 0
+    buf = UInt8[]
     while !eof(from)
-        n += write(to, readavailable(from))
+        nb = Int(clamp(bytesavailable(from), 1, 2^24))
```

Comment:
1. `bytesavailable` does not have a fallback implementation. While it's not good style, I would use `hasmethod` to check and otherwise just read a fixed size
2. 2^24 does seem like a lot. Normal buffer sizes are in the low kilobytes - 8 KiB in Rust, for example. Though I'm not sure - apparently modern hard drives are much more efficient with large reads.